Src/IronPython/Compiler/Ast/PythonAst.cs

```diff
@@ -752,6 +752,10 @@ internal class LookupVisitor : MSAst.ExpressionVisitor {
                     return base.VisitExtension(new GeneratorExpression((FunctionDefinition)VisitScope(generator.Function), generator.Iterable));
                 }
 
+                if (node is Comprehension comprehension) {
+                    return VisitComprehension(comprehension);
```

Comment:
Any reason for putting the `base.VisitExtension` inside `VisitComprehension` instead of here like is done with `VisitScopes`? I'm not particularly concerned about it, the difference just caught my eye.

Reply:
Because a scope comprehension does not "actually show up in the AST hierarchy and instead hangs off the comprehension expression", to quote the comment from the source code.

It has to be this way so that the comprehension (and all its expressions, including e.g. a generator item) will be rewritten within the comprehension scope, rather than the parent scope.

Oddities like this would have been ironed out with a redesign of the inheritance hierarchy.